tests/test_parseCalibsGen2.py

```diff
@@ -0,0 +1,167 @@
+# (http://www.lsst.org).
+# See the COPYRIGHT file at the top-level directory of this distribution
+# for details of code ownership.
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+"""Unit tests for gen2 DECam calibration product parsing, a precursor to ingestion.
+"""
+
+import unittest
+import os
+import lsst.utils.tests
+from lsst.obs.decam import ingestCalibs
+import lsst.daf.persistence as dafPersist
+from lsst.afw.fits import readMetadata
+
+testDataPackage = "testdata_decam"
+try:
+    testDataDirectory = lsst.utils.getPackageDir(testDataPackage)
+except LookupError:
+    testDataDirectory = None
+
+
+@unittest.skipIf(testDataDirectory is None, "testdata_decam must be set up")
+class DecamCalibsParseTaskTestCase(lsst.utils.tests.TestCase):
+    """Test the methods in DecamCalibsParseTask.
+
+    Uses a bias+flat pair of LSST-Science-Pipeline-built calibs as well as
+    a bias+flat pair of DECam Community Pipeline calibs, all full of zeros,
+    which live in testdata_decam.
+    """
+    def setUp(self):
+        self.task = ingestCalibs.DecamCalibsParseTask(name='ItsATest')
+
+        # LSST-Science-Pipeline-built calibration products, full of zeros
+        self.biasFile = os.path.join(testDataDirectory, "pipeline-calibs", "BIAS", "BIAS-47.fits")
+        self.flatFile = os.path.join(testDataDirectory, "pipeline-calibs", "FLAT", "FLAT-g-47.fits")
+
+        # DECam Community Pipeline calibration products, also full of zeros
+        cpBiasFilename = "c4d_150218_191721_zci_v1.fits.fz"
+        self.cpBiasFile = os.path.join(testDataDirectory, "hits2015-zeroed", cpBiasFilename)
+        cpFlatFilename = "c4d_150218_200522_fci_g_v1.fits.fz"
+        self.cpFlatFile = os.path.join(testDataDirectory, "hits2015-zeroed", cpFlatFilename)
+
+        # A Butler repository with ingested raws and Community Pipeline calibs
+        repo = os.path.join(testDataDirectory, "ingested")
+        calibRepo = os.path.join(testDataDirectory, "calibingested")
+        self.butler = dafPersist.Butler(root=repo, calibRoot=calibRepo)
+
+    def testLsstCalibsGetInfo(self):
+        """Test that the filepath and a calib_hdu value of 1 is in the info.
+        """
+        # LSST-Science-Pipelines-built bias
+        phuInfo, infoList = self.task.getInfo(self.biasFile)
+        assert phuInfo['calib_hdu'] == 1
+        assert 'path' in phuInfo
```

Comment:
All of these `assert`s should be turned into `self.assertX` statements (for `X=` e.g. `Equal`, `In`, etc.).

https://developer.lsst.io/python/testing.html